.github/workflows/dependabot-tidy.yml

```
@@ -0,0 +1,42 @@
+name: Tidy All Go Modules
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - '**/go.mod'
+      - '**/go.sum'
+
+jobs:
+  tidy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+
+      - name: Set up Go
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 #v5.5.0
+        with:
+          go-version-file: "go.mod"
+
+      - name: Run go mod tidy in all modules
+        run: |
+          find . -name 'go.mod' -execdir sh -c '
+            echo "Running go mod tidy in $(pwd)"
+            go mod tidy
+          ' \;
+
+      - name: Commit and push tidy changes (only Dependabot PRs)
+#        if: github.event_name == 'pull_request' && github.actor == 'dependabot[bot]'
```

[question] should this be uncommented?
```suggestion
        if: github.event_name == 'pull_request' && github.actor == 'dependabot[bot]'
```

---

previously tried to create commits to stuff from ci in the distant past but gave up

recently closed #5455 suspecting it would have the same issues

worth a try tho!